coriolis/osmorphing/base.py

```diff
@@ -393,39 +394,78 @@ def _set_selinux_autorelabel(self):
         except Exception as err:
             LOG.warning("Failed to set autorelabel: %r" % err)
 
-    def _configure_cloud_init_user_retention(self):
-        cloud_cfg_paths = ["/etc/cloud/cloud.cfg"]
+    def _write_cloud_init_mods_config(self, cloud_cfg):
+        cloud_config_path = "/etc/cloud/cloud.cfg.d/99_coriolis.cfg"
         cloud_cfgs_dir = "/etc/cloud/cloud.cfg.d"
-        if self._test_path(cloud_cfgs_dir):
-            for path in self._exec_cmd_chroot(
-                    'ls -1 %s' % cloud_cfgs_dir).decode().splitlines():
-                if path.endswith('.cfg'):
-                    cloud_cfg_paths.append("%s/%s" % (cloud_cfgs_dir, path))
-
+        if not self._test_path(cloud_cfgs_dir):
+            exception.CoriolisException("Could not find %s" % cloud_cfgs_dir)
         self._event_manager.progress_update(
-            "Reconfiguring cloud-init to retain original user credentials")
-        for cloud_cfg_path in cloud_cfg_paths:
-            if not self._test_path(cloud_cfg_path):
-                LOG.warn("Could not find %s. Skipping reconfiguration." % (
-                    cloud_cfg_path))
-                continue
+            "Configuring cloud-init additional options")
+        new_cloud_cfg = yaml.dump(cloud_cfg)
+        self._write_file_sudo(cloud_config_path, new_cloud_cfg)
+
+    def _disable_installer_cloud_config(self):
+        installer_config_path = "/etc/cloud/cloud.cfg.d/99-installer.cfg"
+        if self._test_path(installer_config_path):
+            self._event_manager.progress_update(
+                "Disabling Ubuntu installer-generated cloud-config")
+            self._exec_cmd_chroot(
+                f"mv {installer_config_path} {installer_config_path}.bak")
+
+    def _ensure_cloud_init_not_disabled(self):
+        disabler_file = "/etc/cloud/cloud-init.disabled"
+        system_conf_disabler = "/etc/systemd/system.conf"
+        grub_conf_disabler = "/etc/default/grub"
+        if self._test_path(disabler_file):
+            self._exec_cmd_chroot(f"rm {disabler_file}")
+        if self._test_path(system_conf_disabler):
+            self._exec_cmd_chroot(
+                "sed -i '/cloud-init=disabled/d' %s" % system_conf_disabler)
+        if self._test_path(grub_conf_disabler):
+            self._exec_cmd_chroot(
+                "sed -i '/cloud-init=disabled/d' %s" % grub_conf_disabler)
```

Comment:
This I know for sure will not work, because any grub update needs an `update-grub` or equivalent (redhat-derivates have a different command). 

In some cases, this grub update may not happen, so we need to also run the grub update command here.

I think for both cases (grub and system.conf), you'll need to first `grep` and then do sed and/or update commands.